planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -84,7 +84,7 @@
     // Each DirectionsLeg contains starting address, ending address, and duration
     // DirectionRoute also has int[] waypointOrder, which will contain the order of the waypoints
     // For the MVP, real time traffic is NOT used.
-    private DirectionsRoute getDirectionsRoute(List<Event> events, boolean optimized) throws ItineraryException{
+    public DirectionsRoute getDirectionsRoute(List<Event> events, boolean optimized) throws ItineraryException{
```

Comment:
Instead of making a method public just so that its own test can mock it, we can have a helper class as a thin wrapper around Maps API. We pass the helper class into ItineraryGenerator in its constructor and mock the helper in tests.
The pattern is called [_dependency injection_](https://en.wikipedia.org/wiki/Dependency_injection) in case you want to read more about it.
I recommend you refactoring the code if you have time in your last week.

Reply:
Ah okay, thank you!

Reply:
Also consider replace the boolean |optimized| with your enum in this file

Reply:
Fixed!